store/sql_channel_store.go

```diff
@@ -405,7 +405,7 @@ func (s SqlChannelStore) get(id string, master bool, allowFromCache bool) StoreC
 				if metrics != nil {
 					metrics.IncrementMemCacheHitCounter("Channel")
 				}
-				result.Data = cacheItem.(*model.Channel)
+				result.Data = (cacheItem.(*model.Channel)).DeepCopy()
```

Comment:
getting the item from the cache is atomic and anything in the cache is suppose to be read only.  The underlying cache object may change but you don't care at that point since you're working with the old data.